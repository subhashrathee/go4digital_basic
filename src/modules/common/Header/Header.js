import React from 'react'
import { NavLink } from 'react-router-dom'

/**
* Header component which contains the site menu and links.
*
* Props:      None
* 
* State vars: None
*/
class Header extends React.Component {
	render() { 
		return (
  <header>
    <div id="wrap">
          <NavLink exact={true}  to="/">
            <img className="logo" src="/img/logo.png" alt="Logo"/>
          </NavLink>
          <nav id="nav">
          <ul>
            <li><NavLink exact={true}  to="/" activeClassName="selected">go4digital</NavLink></li>
            <li><NavLink to="/services/" activeClassName="selected">Services</NavLink></li>
            <li><NavLink to="/about_us/" activeClassName="selected">About</NavLink></li>
            <li><NavLink to="/blog/" activeClassName="selected">Blog</NavLink></li>
            <li><NavLink to="/contact_us/" activeClassName="selected">Contact</NavLink></li>
            <li><a href="https://www.instagram.com/go4digital1/" title="Instagram" target="_blank" rel="noopener noreferrer">
              <img className="instagram-header-icon" src="/img/instagram-header-icon.png" alt="Instagram Icon" /></a></li>
            
            <li><a href="https://www.facebook.com/Go4digital-964819073705597/" title="go4digital at Facebook" target="_blank" rel="noopener noreferrer">
              <img className="instagram-header-icon" src="/img/FB-f-Logo__blue_29.png" alt="Facebook Icon" /></a></li>
            
            
          </ul>
        </nav>
    </div>
  </header>
		)
	}
}

export default Header;
