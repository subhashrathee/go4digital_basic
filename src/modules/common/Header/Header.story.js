import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { BrowserRouter } from 'react-router-dom'
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs/react';

import '../../../../public/css/external_css.css'

import Header from './Header'

const stories = storiesOf('Storybook Knobs', module);

// Add the `withKnobs` decorator to add knobs support to your stories.
// You can also configure `withKnobs` as a global decorator.
stories.addDecorator(withKnobs);

storiesOf('Header', module)
  .add('Default Header', () => (
		<BrowserRouter>
  		<Header />
		</BrowserRouter>
	));
