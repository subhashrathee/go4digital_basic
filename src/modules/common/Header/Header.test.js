import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';
import { BrowserRouter, Link } from 'react-router-dom'

test('Header component', () => {
  const component = renderer.create(
    <BrowserRouter>
		<Header />
		</BrowserRouter>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
