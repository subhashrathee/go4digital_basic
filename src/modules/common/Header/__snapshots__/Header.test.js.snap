// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component 1`] = `
<header>
  <div
    id="wrap"
  >
    <a
      aria-current="true"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="Logo"
        className="logo"
        src="/img/logo.png"
      />
    </a>
    <nav
      id="nav"
    >
      <ul>
        <li>
          <a
            aria-current="true"
            className="selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            go4digital
          </a>
        </li>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/services/"
            onClick={[Function]}
            style={undefined}
          >
            Services
          </a>
        </li>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/about_us/"
            onClick={[Function]}
            style={undefined}
          >
            About
          </a>
        </li>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/blog/"
            onClick={[Function]}
            style={undefined}
          >
            Blog
          </a>
        </li>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/contact_us/"
            onClick={[Function]}
            style={undefined}
          >
            Contact
          </a>
        </li>
        <li>
          <a
            href="https://www.instagram.com/go4digital1/"
            rel="noopener noreferrer"
            target="_blank"
            title="Instagram"
          >
            <img
              alt="Instagram Icon"
              className="instagram-header-icon"
              src="/img/instagram-header-icon.png"
            />
          </a>
        </li>
        <li>
          <a
            href="https://www.facebook.com/Go4digital-964819073705597/"
            rel="noopener noreferrer"
            target="_blank"
            title="go4digital at Facebook"
          >
            <img
              alt="Facebook Icon"
              className="instagram-header-icon"
              src="/img/FB-f-Logo__blue_29.png"
            />
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
